PMNS_NSIs.cxx

```diff
@@ -0,0 +1,409 @@
+    cout << "Setting reverse order (Eta_" << flvj << flvi << "). " << endl;
+    int temp = flvi;
+    flvi = flvj;
+    flvj = temp;
+  }
+  if(flvi<0 || flvi>2 || flvj < flvi || flvj > 2){
+    cout << "Eta_" << flvi << flvj << " not valid for " << fNumNus;
+    cout << " neutrinos. Returning 0." << endl;
+    return zero;
+  }
+
+  return fEta[flvi][flvj];
+
+}
+
+void PMNS_NSIs::SetLowestMass(double m){
+
+double h=m;
+fM=h;
+
+}
+//......................................................................
+///
+/// Set eta_ee parameter
+///
+/// This will check if value is changing to keep track of whether
+/// the eigensystem needs to be recomputed.
+///
+/// @param a - The value of the real parameter eta_ee
+///
+void PMNS_NSIs::SetEta_ee(double a){ SetEta(0,0, a, 0); }
+
+//......................................................................
+///
+/// Set eta_mumu parameter
+///
+/// This will check if value is changing to keep track of whether
+/// the eigensystem needs to be recomputed.
+///
+/// @param a - The value of the real parameter eta_mumu
+///
+void PMNS_NSIs::SetEta_mumu(double a){ SetEta(1,1, a, 0); }
+
+//......................................................................
+///
+/// Set eta_tautau parameter
+///
+/// This will check if value is changing to keep track of whether
+/// the eigensystem needs to be recomputed.
+///
+/// @param a - The value of the real parameter eta_tautau
+///
+void PMNS_NSIs::SetEta_tautau(double a){ SetEta(2,2, a, 0); }
+
+//......................................................................
+
+void PMNS_NSIs::SetEta_emu(double a, double phi){ SetEta(0,1, a, phi); }
+
+//......................................................................
+///
+/// Set eta_mumu parameter
+///
+/// This will check if value is changing to keep track of whether
+/// the eigensystem needs to be recomputed.
+///
+/// @param a - The value of the real parameter eta_mumu
+///
+void PMNS_NSIs::SetEta_mutau(double a, double phi){ SetEta(1,2, a, phi); }
+
+//......................................................................
+///
+/// Set eta_tautau parameter
+///
+/// This will check if value is changing to keep track of whether
+/// the eigensystem needs to be recomputed.
+///
+/// @param a - The value of the real parameter eta_tautau
+///
+void PMNS_NSIs::SetEta_etau(double a, double phi){ SetEta(0,2, a, phi); }
+
+
+////Set lightest neutrino mass
+void PMNS_NSIs::SetM(double m){ SetLowestMass(m);}
+
+double PMNS_NSIs::GetM(){ return fM;}
+/// Set eps_mutau parameter
+///
+/// This will check if value is changing to keep track of whether
+/// the eigensystem needs to be recomputed.
+///
+/// @param a   - The absolute value of the parameter eps_mutau
+///
+/// Build the full Hamiltonian in matter.
+///
+/// Here we divide the mass squared matrix Hms by the 2E
+/// to obtain the vacuum Hamiltonian in eV. Then, the matter
+/// potential is added to the electron component.
+///
+void PMNS_NSIs::UpdateHam()
```

Comment:
This function needs a lot of style improvement. I'd start by making sure indentation is properly done. Use two spaces for each indentation level.

Reply:
Made proper identation. Also added documentation comments so that every task inside the loop can be identified.

Reply:
I still see some indentation issues